### PR TITLE
gensync: new port

### PR DESCRIPTION
**Pull request by Chenxy517**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 13.0.1 22A400 arm64
Xcode 14.3.1 14E300c

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: net/gensync/Portfile [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+PortGroup               github 1.0
+
+github.setup            nislab gensync 1.0.4 v
+git.branch              vanilla
+
+name                    gensync
+version                 1.0
+revision                0
+
+categories              net dev
```
Comment (herbygillot) [inline net/gensync/Portfile:14]: Did you mean `devel` instead of `dev`?

File: net/gensync/Portfile [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+PortSystem              1.0
+PortGroup               cmake 1.1
+PortGroup               github 1.0
+
+github.setup            nislab gensync 1.0.4 v
+git.branch              vanilla
+
+name                    gensync
+version                 1.0
```
Comment (herbygillot) [inline net/gensync/Portfile:11]: Why are you manually setting the version to `1.0` when it appears that it should correctly be `1.0.4` as according to the `github.setup` line?

File: net/gensync/Portfile [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               cmake 1.1
+PortGroup               github 1.0
+
+github.setup            nislab gensync 1.0.4 v
+git.branch              vanilla
+
+name                    gensync
```
Comment (herbygillot) [inline net/gensync/Portfile:10]: `name` is already set by `github.setup`.  You can delete this line.

File: net/gensync/Portfile [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+version                 1.0
+revision                0
+
+categories              net dev
+platforms               darwin
```
Comment (herbygillot) [inline net/gensync/Portfile:15]: `platforms darwin` is now implicit, you do not need this line.

File: net/gensync/Portfile [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+license                 GPL-3
+maintainers             {@trachten bu.edu:trachten} openmaintainer
+
+description             Gensync: a library for efficient synchronization of data over a network
+long_description        Gensync is a library that uses many different syncing \
+                        algorithms to sync data between two nodes in a network. \
+                        These algorithms include IBLTs, CPISyncs, HashSyncs, Cuckoo Syncs, and more.
+
+homepage                https://github.com/nislab/gensync/
```
Comment (herbygillot) [inline net/gensync/Portfile:24]: `homepage` is implicitly set by `github.setup`. You can delete this line.

File: net/gensync/Portfile [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+fetch.type              git
+
+post-fetch {
+    system -W ${worksrcpath} "git submodule update --init --recursive"
+}
+
+checksums               rmd160  83d1555d700fc4a7f1259bf7630f30de43fcade4 \
```
Comment (herbygillot) [inline net/gensync/Portfile:32]: Since you are using `fetch.type git`, MacPorts will not use `checksums`.  You can remove this `checksums` block.

**Comment by herbygillot** [comment]
Thank you for this submission, @Chenxy517.

As per MacPorts policy, please squash all changes into a single commit whose commit message looks like `<port name>: <description>`:

In this case, it would be `gensync: new port`

Thanks.

**Comment by Chenxy517** [comment]
Thank you for the review, @herbygillot.

I've fixed all the issues.

File: net/gensync/Portfile [hunk 7]
```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               cmake 1.1
+PortGroup               github 1.0
+
+github.setup            nislab gensync 1.0.4 v
+git.branch              vanilla
+
+version                 1.0.4
```
Comment (herbygillot) [inline net/gensync/Portfile:10]: I apologize - I should have also mentioned that `version` is also being implicitly set by `github.setup`; this line is redundant and may be removed.

File: net/gensync/Portfile [hunk 8]
```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               cmake 1.1
+PortGroup               github 1.0
+
+github.setup            nislab gensync 1.0.4 v
+git.branch              vanilla
```
Comment (herbygillot) [inline net/gensync/Portfile:8]: Is it necessary to manually set the branch to `vanilla`?  If you do not set this, MacPorts will just use the `v1.0.4` git tag, which is usually what's expected.

Comment (Chenxy517) [reply net/gensync/Portfile]: `git.banch` has now been removed. Thanks

**Comment by herbygillot** [comment]
Thank you for your submission, and for addressing all comments, @Chenxy517.

**Comment by herbygillot** [comment]
Unfortunately `gensync` is failing to build in the MacPorts buildbots. Here is the log: https://build.macports.org/builders/ports-13_x86_64-builder/builds/40784/steps/install-port/logs/stdio

**Comment by kencu** [comment]
likely to be a case issue I would think.

**Pull request merged**